### PR TITLE
fix: introduce PR groups for dependabot

### PR DESCRIPTION
**Pull request by wtrocki**
Just an idea to help us with more frequent and efficient PR reviews.
NOTE: Dependabot is quite limited in that regard. Renovate is much more powerful and also free.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -7,6 +7,25 @@ updates:
       day: tuesday
     reviewers:
       - "mongodb/mongocli"
+    groups:
+      internal-dependencies:
+          patterns:
+            - "go.mongodb.org*" # A single dependency name
```
Comment (wtrocki) [inline .github/dependabot.yml:13]: Those should have the highest priority and can be labeled differently for help rota actions

**Comment by wtrocki** [comment]
FYI @gssbzn WDYT about trying this out (add remove some groups). 
If that is going to clash with existing owners json and processes  we can close the PR.

File: .github/dependabot.yml (hunk 1 above)
Comment (gssbzn) [reply .github/dependabot.yml:13]: this will include the go driver which is kind of 🤷 for us

**Review by gssbzn — APPROVED** [review]
:shipit: and let's see how it goes

File: .github/dependabot.yml (hunk 1 above)
Comment (wtrocki) [reply .github/dependabot.yml]: Removing section. It will introduce more harm than good.

**Review by gssbzn — APPROVED** [review]

**Pull request merged**